daemon/mgr/container.go

```diff
@@ -607,9 +602,18 @@ func (mgr *ContainerManager) Start(ctx context.Context, id string, options *type
 }
 
 func (mgr *ContainerManager) start(ctx context.Context, c *Container, options *types.ContainerStartOptions) error {
+	// NOTE: add a big lock when start a container
+	c.Lock()
+	defer c.Unlock()
+
 	var err error
 	c.DetachKeys = options.DetachKeys
 
+	// check if container's status is running
+	if c.State.Running {
+		return errors.Wrapf(errtypes.ErrNotModified, "container already started")
```

Comment:
I check docker start a running state container should not throw error, why we here

Reply:
Actually no specific reasons, we can make the result staying with the moby

Reply:
Sure

Reply:
> I check docker start a running state container should not throw an error, why we here

which docker version do you check? I check the lastest moby and 1.12.6 docker, they all throw an error when starting an already running container.

Reply:
docker 1.12.6 version
```
[root@r10e19288.sqa.zmf /root]
#docker start caff7e06c83d
caff7e06c83d

[root@r10e19288.sqa.zmf /root]
#docker start caff7e06c83d
caff7e06c83d
```

Reply:
Oh, I know, that error not throw in cli 
```
time="2019-03-18T09:52:13.025216482+08:00" level=info msg="Calling POST /v1.24/containers/caff7e06c83d/start, client @"
time="2019-03-18T09:52:13.025397258+08:00" level=error msg="Handler for POST /v1.24/containers/caff7e06c83d/start, client @ returned error: Container already started"
```